pkg/ipmi/oem.go

```diff
@@ -0,0 +1,298 @@
+// Copyright 2017-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ipmi
+
+import (
+	"bytes"
+	"encoding/binary"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	"github.com/u-root/u-root/pkg/smbios"
+)
+
+const (
+	_IPMI_FB_OEM_NET_FUNCTION = 0x36
+
+	//OEM Commands
+	_FB_OEM_SET_PROC_INFO = 0x10
+	_FB_OEM_SET_DIMM_INFO = 0x12
+)
+
+type ProcessorInfo struct {
+	ManufacturerID        [3]uint8
+	Index                 uint8
+	ParameterSelector     uint8
+	ProductName           [48]byte
+	CoreNumber            uint8
+	ThreadNumberLSB       uint8
+	ThreadNumberMSB       uint8
+	ProcessorFrequencyLSB uint8
+	ProcessorFrequencyMSB uint8
+	Revision1             uint8
+	Revision2             uint8
+}
+
+type DimmInfo struct {
+	ManufacturerID          [3]uint8
+	Index                   uint8
+	ParameterSelector       uint8
+	DIMMPresent             uint8
+	NodeNumber              uint8
+	ChannelNumber           uint8
+	DIMMNumber              uint8
+	DIMMType                uint8
+	DIMMSpeed               uint16
+	DIMMSize                uint32
+	ModulePartNumber        [20]byte
+	ModuleSerialNumber      uint32
+	ModuleManufacturerIDLSB uint8
+	ModuleManufacturerIDMSB uint8
+}
+
+var BoardManufacturer = map[string][3]uint8{
+	"Wiwynn": {0x0, 0x9c, 0x9c},
+}
+
+func (i *IPMI) SendOemIpmiCommand(netfn byte, cmd byte, data []byte) error {
```

Comment:
Can we have an function that takes the ProcessorInfo and sends it and another one that takes the DimmInfo and sends it?

Somehting like `SendOemIpmiProcessorInfo` and `SendOemIpmiDimmInfo` to avoid having to pass a ntfn byte and a cmd byte, and have a higher level astraction.

Reply:
Done.